File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/maneger/service/BrandService.java
package com.pyg.maneger.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/8/10.
 */
public interface BrandService {
    /**
     * 需求：查询所有品牌数据
     */
    public List<TbBrand> findAll();

    //分页查询
    PageResult findPage(int pageNum, int pageSize);
    //保存品牌
    void save(TbBrand tbBrand);
    //根据id查询
    TbBrand findById(Long id);
    //根据id修改
    void updata(TbBrand tbBrand);
    //根据id删除
    void del(Long [] ids);
    //查询品牌
    List<Map> findbrandList();
}

File: pyg-shop-web/src/main/webapp/js/service/brandService.js
//定义模块函数
app.service('brandService',function ($http) {
    //查询全部
    this.findAll=function () {
        return  $http.get('../brand/findAll')
    };
    //分页查询
    this.findPage=function (pageNum, pageSize) {
        return	$http.get('../brand/findPage/'+pageNum+'/'+pageSize);
    };
    //根据ID查询
    this.findById=function (id) {
        return $http.get("../brand/findById/"+id)
    };
    //保存与修改操作
    this.save=function (entity) {
        return $http.post("../brand/save/",entity);
    };
    this.updata=function (entity) {
        return   $http.post("../brand/updata/",entity);
    };
    //删除操作
    this.del=function (ids) {
        return	$http.get('../brand/del/'+ids);
    }
    this.findbrandList=function () {
        return $http.get('../brand/findbrandList');
    }
});